src/include/kdbprivate.h

```diff
@@ -539,6 +539,54 @@ int elektraKeyLock (Key * key, enum elektraLockOptions what);
 
 ssize_t ksSearchInternal (const KeySet * ks, const Key * toAppend);
 
+/**
+ * The Vstack structure. Used in the OPMPHM.
+ *
+ * A stack implementation with dynamical memory allocation.
+ * The space gets doubled if full and reduced by half if used only a quarter of it,
+ * but not less than minSize.
+ */
+
+typedef struct
+{
+	int minSize;  /*!< the minimal size of the stack > */
+	int size;     /*!< The maximal size of the stack */
+	void ** data; /*!< The data array */
+	void ** head; /*!< The stack pointer, which points to the head */
+} Vstack;
+
+Vstack * elektraVstackInit (int minSize);
+int elektraVstackPush (Vstack * stack, void * data);
+void * elektraVstackPop (Vstack * stack);
+int elektraVstackIsEmpty (Vstack * stack);
+void elektraVstackDestroy (Vstack * stack);
+
+/**
+ * The Vheap structure. Used in the OPMPHM.
```

Comment:
Please explain what a VHeap is, its not so important where it is used (should be in detail and not in brief)